homeassistant/components/nexia/services.py

```diff
@@ -0,0 +1,58 @@
+"""Services for nexia."""
+
+import voluptuous as vol
+
+from homeassistant.components.climate.const import ATTR_HUMIDITY
+from homeassistant.const import ATTR_ENTITY_ID
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+from .const import (
+    ATTR_AIRCLEANER_MODE,
+    DOMAIN,
+    SIGNAL_AIRCLEANER_SERVICE,
+    SIGNAL_HUMIDIFY_SERVICE,
+)
+
+SERVICE_SET_AIRCLEANER_MODE = "set_aircleaner_mode"
+SERVICE_SET_HUMIDIFY_SETPOINT = "set_humidify_setpoint"
+
+SET_FAN_MIN_ON_TIME_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_AIRCLEANER_MODE): cv.string,
+    }
+)
+
+SET_HUMIDITY_SCHEMA = vol.Schema(
+    {
+        vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+        vol.Required(ATTR_HUMIDITY): vol.All(
+            vol.Coerce(int), vol.Range(min=35, max=65)
+        ),
+    }
+)
+
+
+def register_climate_services(hass):
+    """Register all climate services for nexia."""
+
+    async def _humidify_set_service(service):
+        async_dispatcher_send(hass, SIGNAL_HUMIDIFY_SERVICE, service.data)
+
+    hass.services.async_register(
+        DOMAIN,
+        SERVICE_SET_HUMIDIFY_SETPOINT,
+        _humidify_set_service,
+        schema=SET_HUMIDITY_SCHEMA,
+    )
+
+    async def _aircleaner_set_service(service):
+        async_dispatcher_send(hass, SIGNAL_AIRCLEANER_SERVICE, service.data)
```

Comment:
Calling a service should wait until the request is done. By sending a dispatcher, the service task will return right away before the service call is actually executed.

Reply:
@balloob I'm kinda stuck with this approach due to https://github.com/home-assistant/core/pull/33078#issuecomment-602007549

Reply:
Oh shit yes. Let me see if I can whip something up.

Reply:
Converted :)